openslides/utils/autoupdate.py

```diff
@@ -253,6 +269,44 @@ def inform_deleted_data(*args, information=None):
     transaction.on_commit(lambda: send_autoupdate(collection_elements))
 
 
+def inform_changed_permissions(group, given_permissions):
+        """
+        This function informs the autoupdate system about changed permissions
+        and manage the necessary data to be send to affected clients.
+
+        given_permissions is a list of permissions retrieved from the client. First
+        the difference between these permissions and the old group permissions is
+        determinated to get the new (added) permissions.
+        Second all apps are queried to give the new required collections. These
+        are send out to all members of the group.
+        """
+        old_permissions = group.permissions.all()
```

Comment:
I would like to have permission related code inside user app, not here. The code in the user app can do the signal stuff and then call a short helper here.

I once more suggest to use inform_changed_data because we gain the support of the extra channel way (internal channel 'autoupdate.send_data') and can reuse all code without repeating ourselves).

The diff is ok. Maybe you can describe the diff function a little bit more because `first-second` is not much intuitive. (Ah, I think the problem is that you mean `diff = first – second`. I didn't realize this. Also if first is 0 and second is 2, we do not get a result like –2, do we?)

Then please check not to use too much database queries. Use CollectionElement.from_values() instead of Model.objects.get() and thelike.